otelcollector/test/utils/kubernetes_api_utils.go

```diff
@@ -306,6 +306,62 @@ func CheckIfAllContainersAreRunning(clientset *kubernetes.Clientset, namespace,
 	return nil
 }
 
+/*
+ * Check that pods are scheduled in all the nodes. If a node has no schduled pod on it, return an error.
+ */
+func CheckIfAllPodsScheduleOnNodes(clientset *kubernetes.Clientset, namespace, labelKey string, labelValue string) error {
+
+	// Get list of all nodes
+	nodes, err := clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+
+	if err != nil {
+		return errors.New(fmt.Sprintf("Error getting nodes with the specified labels: %v", err))
+	}
+
+	for _, node := range nodes.Items {
+
+		// Get list of pods scheduled on this node
+		pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
+			FieldSelector: "spec.nodeName=" + node.Name,
+		})
+
+		if err != nil || pods == nil || len(pods.Items) == 0 {
+			return errors.New(fmt.Sprintf("Error getting pods on node %s:", node.Name))
+		}
+	}
+
+	return nil
+}
+
+/*
+ * Check that pods are scheduled in all the Fips nodes. If a Fips node has no schduled pod on it, return an error.
+ */
+func CheckIfAllFipsPodsScheduleOnNodes(clientset *kubernetes.Clientset, namespace, labelKey string, labelValue string) error {
+
+	// Get list of all nodes
+	nodes, err := clientset.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+
+	if err != nil {
+		return errors.New(fmt.Sprintf("Error getting nodes with the specified labels: %v", err))
+	}
+
+	for _, node := range nodes.Items {
+		if value, ok := node.Labels["agentpool"]; ok && value == "fipslin" {
```

Comment:
Is there a different way we can check if this is a fips node other than the agentpool name? This won't work on other clusters unless the fips nodepool is named "fipslin"

Reply:
Thanks Grace! I have updated this to kubernetes.azure.com/fips_enabled = true to be consistent across all environments.